ui/components/multichain/account-details/account-details.js

```diff
@@ -0,0 +1,203 @@
+import React, { useCallback, useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import Popover from '../../ui/popover/popover.component';
+import {
+  setAccountDetailsAddress,
+  setAccountLabel,
+  clearAccountDetails,
+  exportAccount,
+  hideWarning,
+} from '../../../store/actions';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarAccountVariant,
+  BUTTON_SECONDARY_SIZES,
+  BannerAlert,
+  ButtonPrimary,
+  ButtonSecondary,
+  PopoverHeader,
+  Text,
+  TextField,
+} from '../../component-library';
+import Box from '../../ui/box/box';
+import EditableLabel from '../../ui/editable-label/editable-label';
+import {
+  getMetaMaskAccountsOrdered,
+  getMetaMaskKeyrings,
+} from '../../../selectors';
+import QrView from '../../ui/qr-code';
+import { isHardwareKeyring } from '../../../helpers/utils/hardware';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventKeyType,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  DISPLAY,
+  FLEX_DIRECTION,
+  JustifyContent,
+  Size,
+  TextColor,
+  TextVariant,
+  SEVERITIES,
+} from '../../../helpers/constants/design-system';
+import { AddressCopyButton } from '../address-copy-button';
+
+export const AccountDetails = ({ address }) => {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+  const useBlockie = useSelector((state) => state.metamask.useBlockie);
+  const keyrings = useSelector(getMetaMaskKeyrings);
+  const accounts = useSelector(getMetaMaskAccountsOrdered);
+  const { name } = accounts.find((account) => account.address === address);
+  const keyring = keyrings.find((kr) => kr.accounts.includes(address));
+  const exportPrivateKeyFeatureEnabled = !isHardwareKeyring(keyring?.type);
+
+  const [attemptingExport, setAttemptingExport] = useState(false);
+  const [password, setPassword] = useState('');
+
+  // Password error would result from appState
+  const warning = useSelector((state) => state.appState.warning);
+
+  // This is only populated when the user properly authenticates
+  const privateKey = useSelector(
+    (state) => state.appState.accountDetail.privateKey,
+  );
+
+  const onClose = useCallback(() => {
+    dispatch(setAccountDetailsAddress(''));
+    dispatch(clearAccountDetails());
+  }, [dispatch]);
+
+  const onSubmit = useCallback(() => {
+    dispatch(exportAccount(password, address)).then((res) => {
+      dispatch(hideWarning());
+      return res;
+    });
+  }, [dispatch, password, address]);
+
+  return (
+    <Popover
+      headerProps={{
+        paddingBottom: 1,
+      }}
```

Comment:
Can we add a className here so that it becomes easy for other devs to look for this component in future?

Reply:
Let's avoid adding `className` just to find components.  Do you mean a `data-test-id` or something?

Reply:
Yeah, data-test-id will be fine as well.